src/components/ProjectSearch/index.js

```diff
@@ -20,84 +23,266 @@ import {
   getTextSearchQuery
 } from "../../selectors";
 
+import Svg from "../shared/Svg";
+import ManagedTree from "../shared/ManagedTree";
+import SearchInput from "../shared/SearchInput";
+
+import type { List } from "immutable";
+import type { Location } from "../../types";
+import type { ActiveSearchType } from "../../reducers/types";
+import type { StatusType } from "../../reducers/project-text-search";
+
 import "./ProjectSearch.css";
 
+type Match = {
+  type: "MATCH",
```

Comment:
To make `renderItem` work easily adding a type makes a disjoint union easier. Otherwise we'd have compare object members to determine if it's a result or match.